src/pal/src/map/virtual.cpp

```diff
@@ -1096,6 +1096,7 @@ VIRTUALCommitMemory(
         }
 
         StartBoundary = pInformation->startBoundary + runStart * VIRTUAL_PAGE_SIZE;
+        pRetVal = (void *)StartBoundary;
```

Comment:
Welcome to the goto hell in pal, 

pRetVal is assigned unconditionally if we break from the while loop. The only other way to exit the loop is using a goto statetement. while i agree we could be adding a goto done in future, we should be changing pRetVal to null [ or a legitimate value] before we jump. 
Presently we only have a goto error within the while loop and error actually expects the pRetVal to point the page whose commit/change of permission failed, which we are not setting for now. Setting the correct value is what this change is about